File: csharp/src/FizzBuzz.cs
namespace csharp;

public class FizzBuzz : IFizzBuzz
{
    private readonly DivisorAndWord[] _divisorAndWords;

    public FizzBuzz(DivisorAndWord[] divisorAndWords)
    {
        this._divisorAndWords = divisorAndWords;
    }

    public string Say(int number)
    {
        var result = "";
        foreach (var divisorAndWord in _divisorAndWords)
        {
            result += divisorAndWord.GetWordOrEmpty(number);
        }

        if (result != "")
        {
            return result;
        }

        return number.ToString();
    }
}

File: typescript/src/FizzBuzz.ts
import DivisorAndWord from "./DivisorAndWord";
import {IFizzBuzz} from "./IFizzBuzz";

export class FizzBuzz implements IFizzBuzz {
    private readonly divisorAndWords: DivisorAndWord[];

    constructor(divisorAndWords: DivisorAndWord[]) {
        this.divisorAndWords = divisorAndWords;
    }

    public say(inputNumber: number): string {
        let result = "";
        for (const divisorAndWord of this.divisorAndWords) {
            result += divisorAndWord.getWordOrEmpty(inputNumber);
        }

        if (result !== "") {
            return result;
        }

        return String(inputNumber);
    }
}
